src/eos-crash-metrics.c

```diff
@@ -0,0 +1,174 @@
+      return NULL;
+    }
+
+  return sysroot;
+}
+
+static char *
+get_eos_ostree_repo_url (OstreeRepo *repo)
+{
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  char *url = NULL;
+
+  config = ostree_repo_get_config (repo);
+
+  if (!(url = g_key_file_get_string (config, "remote \"eos\"", "url", &error)))
+    {
+      g_warning ("Unable to read OSTree config for eos remote URL: %s", error->message);
+      return NULL;
+    }
+
+  return url;
+}
+
+static char *
+get_eos_ostree_deployment_commit (OstreeSysroot *sysroot, OstreeRepo *repo)
+{
+  OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  GKeyFile *config = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *refspec = NULL;
+  char *commit = NULL;
+
+  if (!deployment)
+    {
+      g_warning ("OSTree deployment is not currently booted, cannot read state");
+      return NULL;
+    }
+
+  config = ostree_deployment_get_origin (deployment);
+
+  if (!(refspec = g_key_file_get_string (config, "origin", "refspec", &error)))
+    {
+      g_warning ("Unable to read OSTree refspec for booted deployment: %s", error->message);
+      return NULL;
+    }
+
+  if (!ostree_repo_resolve_rev (repo, refspec, FALSE, &commit, &error))
+    {
+      g_warning ("Unable to resolve revision for refpsec %s: %s", refspec, error->message);
+      return NULL;
+    }
+
+  return commit;
+}
+
+/* This is the way that the kernel gives us paths ... */
+static const char *blacklisted_prefixes[] =
+{
+  "!home",
+  "!sysroot!home"
+};
+
+static gboolean
+is_blacklisted_path (const char *path)
+{
+  const size_t n_blacklisted_prefixes = G_N_ELEMENTS (blacklisted_prefixes);
+  size_t i = 0;
+
+  for (; i < n_blacklisted_prefixes; ++i)
+    {
+      if (g_str_has_prefix (path, blacklisted_prefixes[i]))
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
+int
+main (int argc, char **argv)
+{
+  const gchar *path = argv[1];
+  const gint16 signal = atoi (argv[2]);
+  const gint64 timestamp = atoll (argv[3]);
+  g_autoptr(OstreeSysroot) sysroot = NULL;
+  g_autoptr(OstreeRepo) repo = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *url = NULL;
+  g_autofree char *commit = NULL;
+
+  if (is_blacklisted_path (path))
+    {
+      g_message ("%s is blacklisted, not reporting crash", path);
+      return EXIT_SUCCESS;
+    }
+
+  sysroot = load_ostree_sysroot ();
+
+  if (ostree_sysroot_get_repo (sysroot, &repo, NULL, &error))
+    {
+      url = get_eos_ostree_repo_url (repo);
+      commit = get_eos_ostree_deployment_commit (sysroot, repo);
+    }
+  else
+    {
+      g_warning ("Unable to read ostree repo from sysroot: %s", error->message);
+      return EXIT_FAILURE;
+    }
+
+  if (!url || !commit)
+    {
+      g_warning ("Unable to get OSTree url or commit, perhaps the system has been tampered with?");
+      return EXIT_FAILURE;
+    }
+
+  report_crash (path, signal, timestamp, url, commit);
```

Comment:
Just found a good one, the report_crash expects url first, then commit. Will fix it up.